stellargraph/core/element_data.py

```diff
@@ -303,74 +303,117 @@ def _numpyise(d, dtype):
     return {k: np.array(v, dtype=dtype) for k, v in d.items()}
 
 
+class AdjList:
+    def __init__(self, init_index):
+        self.init_index = init_index
+        self.undirected = self.ins = self.outs = None
+
+    def force_init(self):
+        self.undirected = self.init_index(ins=True, outs=True)
+        self.ins = self.init_index(ins=True, outs=False)
+        self.outs = self.init_index(ins=False, outs=True)
```

Comment:
Rather than duplicate the `init_index` calling and assignment logic, maybe this could be:

```suggestion
        self.lookup(ins=True, outs=True)
        self.lookup(ins=True, outs=False)
        self.lookup(ins=False, outs=True)
```